packer/plugin.go

```diff
@@ -219,14 +235,19 @@ func (c *PluginConfig) discoverSingle(glob string) (map[string]string, error) {
 	if err != nil {
 		return nil, err
 	}
-
+	prefix := ""
 	res := make(map[string]string)
-
-	prefix := filepath.Base(glob)
-	prefix = prefix[:strings.Index(prefix, "*")]
+	// Sort the matches so we add the newer version of a plugin last
+	sort.Strings(matches)
+	//If we see OS_ARCH (i.e. darwin_arm64 for an m1 mac user) ignore it in the binary name
+	if strings.Contains(glob, OS_ARCH) {
+		prefix = "packer-plugin-"
+	} else {
+		prefix = filepath.Base(glob)
+		prefix = prefix[:strings.Index(prefix, "*")]
+	}
```

Comment:
Unfortunately I don't think that would work because the glob on the install plugins loop has three `*` character's before that, I could check for the index of the last * instead, rather than the first, but the index check would need to be different regardless.  I decided to split it off by OS_ARCH, and assign it as "packer-plugin-", I basically just assume that anything with the OS_ARCH in its binary name is a packer-plugin,

I think this if statement should exist, but I think the condition really should change to a more exact way of detecting if we're looking for installed plugins

Reply:
Ah I got it. I was thinking that `prefix = filepath.Base(glob)` would just give the filename part of the glob leaving you with just 1 asterisk.

Reply:
Oh now I understand, thanks for the follow up clarification, yep you're right! Committing your suggestion now, ty Wilken!